File: leetcode/19/RemoveNode.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dummyHead = new ListNode(0);
        dummyHead->next = head;
        ListNode* front = dummyHead;
        ListNode* end = dummyHead;
        
        for(int i = 0; i < n+1; i++)
        {
            front = front->next;
        }
        
        while(front)
        {
            front = front->next;
            end = end->next;
        }
        
        end->next = end->next->next;
        
        return dummyHead->next;
    }
};

File: leetcode/19/RemoveNode.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @param {number} n
 * @return {ListNode}
 */
 var removeNthFromEnd = function(head, n) {
    let dummyHead = new ListNode(0);
    dummyHead.next = head;
    let front = dummyHead;
    let end = dummyHead;
    
    
    for(let i = 0; i < n+1; i++)
    {
        front = front.next;
    }
    
    while(front != null)
    {
        front = front.next;
        end = end.next;
    }
    
    end.next = end.next.next;
    return dummyHead.next;
    
};